File: src/main/java/me/centralhardware/znatoki/telegram/statistic/telegram/CommandHandler/organization/AddLogChatCommand.java
package me.centralhardware.znatoki.telegram.statistic.telegram.CommandHandler.organization;

import lombok.RequiredArgsConstructor;
import me.centralhardware.znatoki.telegram.statistic.entity.postgres.Role;
import me.centralhardware.znatoki.telegram.statistic.mapper.postgres.OrganizationMapper;
import me.centralhardware.znatoki.telegram.statistic.telegram.CommandHandler.CommandHandler;
import org.springframework.stereotype.Component;
import org.telegram.telegrambots.meta.api.objects.Message;

import java.util.Objects;

@Component
@RequiredArgsConstructor
public class AddLogChatCommand extends CommandHandler {

    private final OrganizationMapper organizationMapper;

    @Override
    public void handle(Message message) {
        if (message.getChatId() > 0){
            sender.sendText("Сначала добавте бота в чат, который хотите использовать для лога, затем выполните команду /join @@OrgStatisticBot ", message.getChatId());
            return;
        }

        var organization = organizationMapper.getByOwner(message.getFrom().getId());

        if (Objects.equals(organization.getLogChatId(), message.getChatId())){
            sender.sendText("Данный чат уже привязан к боту", message.getChatId());
        } else if  (organization.getLogChatId() != null){
            sender.sendText("Чат для логирования уже добавлен", message.getChatId());
        } else {
            organizationMapper.updateLogChat(organization.getId(), message.getChatId());
            sender.sendText("Чат сохранен. Теперь сюда будут приходить уведомления об действиях с ботом от всех сотрудников",message.getChatId());
        }
    }

    @Override
    public boolean isAcceptable(String data) {
        return data.startsWith("/join");
    }

    @Override
    public Role getRequiredRole() {
        return Role.ADMIN;
    }

}

File: src/main/kotlin/me/centralhardware/znatoki/telegram/statistic/telegram/commandHandler/organization/AddLogChatCommand.kt
package me.centralhardware.znatoki.telegram.statistic.telegram.commandHandler.organization

import me.centralhardware.znatoki.telegram.statistic.entity.Role
import me.centralhardware.znatoki.telegram.statistic.mapper.OrganizationMapper
import me.centralhardware.znatoki.telegram.statistic.mapper.UserMapper
import me.centralhardware.znatoki.telegram.statistic.telegram.TelegramSender
import me.centralhardware.znatoki.telegram.statistic.telegram.commandHandler.CommandHandler
import me.centralhardware.znatoki.telegram.statistic.userId
import org.springframework.stereotype.Component
import org.telegram.telegrambots.meta.api.objects.Update

@Component
class AddLogChatCommand(private val organizationMapper: OrganizationMapper,
                        sender: TelegramSender,
                        userMapper: UserMapper
) : CommandHandler(
    sender, userMapper
) {

    override fun handle(update: Update) {
        if (update.userId() > 0L){
            sender.sendText("Сначала добавте бота в чат, который хотите использовать для лога, затем выполните команду /join @@OrgStatisticBot", update.userId())
            return
        }

        val organization = organizationMapper.getByOwner(update.userId())!!

        when {
            organization.logChatId == update.userId() -> sender.sendText("Данный чат уже привязан к боту", update.userId())
            organization.logChatId != null -> sender.sendText("Чат для логирования уже добавлен", update.userId())
            else -> {
                organizationMapper.updateLogChat(organization.id, update.userId())
                sender.sendText("Чат сохранен. Теперь сюда будут приходить уведомления об действиях с ботом от всех сотрудников", update.userId())
            }
        }
    }

    override fun isAcceptable(data: String) = data.startsWith("/join")

    override fun getRequiredRole() = Role.ADMIN
}